remotespark/datawidgets/plotlygraphs/graphrenderer.py

```diff
@@ -13,15 +13,10 @@
 
 class GraphRenderer(object):
 
-    already_initialized = False
-
-    def __init__(self, testing=False):
-        if not testing and not GraphRenderer.already_initialized:
-            init_notebook_mode()
-            GraphRenderer.already_initialized = True
-
     @staticmethod
     def render(df, encoding, output):
+        with output:
```

Comment:
For my own edification, can you describe why this works but the previous solution didn't?